qcodes/interactive_widget.py

```diff
@@ -0,0 +1,436 @@
+                button_kwargs=dict(icon="camera"),
+            )
+            back_button = button(
+                "Back",
+                "warning",
+                on_click=_input_to_button(dct, box),
+                button_kwargs=dict(icon="undo"),
+            )
+            box.children = (
+                text_input,
+                snapshot_button,
+                plot_button,
+                back_button,
+            )
+
+        return on_click
+
+    def _input_to_button(dct: Dict, box: Box) -> Callable[[Button], None]:
+        def on_click(_: Button) -> None:
+            box.children = (_changeable_button(dct, box),)
+
+        return on_click
+
+    def _changeable_button(dct: Dict, box: Box) -> Button:
+        return button(
+            ", ".join(dct), "success", on_click=_button_to_input(dct, box),
+        )
+
+    box = VBox([], layout=Layout(height="auto", width="auto"))
+    box.children = (_changeable_button(dct, box),)
+    return box
+
+
+def _get_coords_and_vars(ds: DataSet) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    coordinates = {}
+    variables = {}
+    for p, spec in ds.paramspecs.items():
+        attrs = {
+            "unit": spec.unit,
+            "label": spec.label,
+            "type": spec.type,
+        }
+        if spec.depends_on:
+            attrs["depends_on"] = spec.depends_on.split(", ")  # type: ignore
+            variables[p] = attrs
+        else:
+            coordinates[p] = attrs
+    return coordinates, variables
+
+
+def _experiment_widget(tab: Tab) -> GridspecLayout:
+    """Show a `ipywidgets.GridspecLayout` with information about the
+    loaded experiment. The clickable buttons can perform an action in ``tab``.
+    """
+    header_names = [
+        "Run ID",
+        "Name",
+        "Experiment",
+        "Coordinates",
+        "Variables",
+        "MSMT Time",
+        "Notes",
+    ]
+
+    header = {n: button(n, "info") for n in header_names}
+    rows = [header]
+    for exp in qcodes.experiments():
+        tooltip = f"{exp.name}#{exp.sample_name}@{exp.path_to_db}"
+
+        for ds in exp.data_sets():
+            coords, variables = _get_coords_and_vars(ds)
+            row = {}
+            row["Run ID"] = label(str(ds.run_id))
+            row["Name"] = label(ds.name)
+            row["Experiment"] = button(
+                f"#{exp.exp_id}", "warning", tooltip=tooltip
+            )
+            row["Notes"] = editable_metadata(ds)
+            row["Coordinates"] = expandable_dict(coords, tab, ds)
+            row["Variables"] = expandable_dict(variables, tab, ds)
+            row["MSMT Time"] = label(ds.completed_timestamp() or "")
+            rows.append(row)
+
+    grid = GridspecLayout(n_rows=len(rows), n_columns=len(header_names))
+
+    empty_label = label("")
+    for i, row in enumerate(rows):
+        for j, name in enumerate(header_names):
+            grid[i, j] = row.get(name, empty_label)
```

Comment:
i tried to use the widget on a large DB with ~300 datasets and i noticed that the order in which datasets appeared was not following the run id, the left most column. So it got me thinking that we'd actually want some kind of sorting. We could leave this for later (because, for example, we could swtich to using SQL queries to populate the widget which might help with the sorting it, where efficient, on sql side).

Reply:
In 1c162386d, I sort Data_Sets on run_id when data_sets isn't explicitly passed.

Reply:
ok, good.

i know from plottr/inspectr experiences that users like the descending sorting by timestamp (newest first). perhaps we should add an option for that? or leave to some later PR?

Reply:
In c1f503ebc, I implement sorting by timestamp, run_id, or None:
![image](https://user-images.githubusercontent.com/6897215/91451832-20adb600-e87e-11ea-9fab-fed92929a9d7.png)